pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -467,19 +467,24 @@ func (r *ReferenceAdapter) deleteCredentials() error {
 	return nil
 }
 
-// ensureAvailibilityZonesSet sets the az in the projectclaim spec if necessary
+// ensureClaimAvailabilityZonesSet sets the az in the projectclaim spec if necessary
 // returns true if the project claim has been modified
-func (r *ReferenceAdapter) ensureClaimAvailibilityZonesSet() (bool, error) {
+func (r *ReferenceAdapter) ensureClaimAvailabilityZonesSet() (bool, error) {
 	if len(r.ProjectClaim.Spec.AvailibilityZones) > 0 {
 		return false, nil
 	}
 
+	if len(r.ProjectClaim.Spec.AvailabilityZones) > 0 {
+		return false, nil
+	}
+
 	zones, err := r.gcpClient.ListAvilibilityZones(r.ProjectReference.Spec.GCPProjectID, r.ProjectClaim.Spec.Region)
 	if err != nil {
 		return false, err
 	}
 
 	r.ProjectClaim.Spec.AvailibilityZones = zones
+	r.ProjectClaim.Spec.AvailabilityZones = zones
```

Comment:
does that mean the AvailabilityZones is a single direction change? meaning from gcp -> AvailibilityZones && AvailabilityZones and no logic is preformed on them?

Reply:
@NautiluX has responed to me privately, so this is resolved